libs/@blockprotocol/graph/src/types/ontology/property-type.ts

```diff
@@ -12,3 +19,29 @@ export type PropertyTypeWithMetadata = {
   schema: PropertyType;
   metadata: OntologyElementMetadata;
 };
+
+export type AggregatePropertyTypesData = {
+  graphResolveDepths?: Partial<
+    Pick<
+      Subgraph<true>["depths"],
+      "constrainsValuesOn" | "constrainsPropertiesOn"
+    >
+  >;
+};
+
+export type AggregatePropertyTypesResult = {
+  results: Subgraph<true, PropertyTypeRootType>;
+};
+
+export type GetPropertyTypeData = {
+  propertyTypeId: VersionedUri;
+};
+
+export type CreatePropertyTypeData = {
+  propertyType: PropertyType;
```

Comment:
In HASH this is represented as
```ts
export type CreatePropertyTypeRequest = {
  propertyType: Omit<PropertyType, "$id">;
};
```

I think we should look at doing the same here. Callers (blocks) _**cannot**_ freely determine type IDs, they have no mechanism of knowing what URLs an EA is capable of serving.

Reply:
Done in https://github.com/blockprotocol/blockprotocol/pull/946/commits/086f506ad7e3054eed4c6896363cce4d53a06145